Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/Processes/ApiGatewayEmulatorProcess.cs

```diff
@@ -0,0 +1,78 @@
+using Amazon.Lambda.TestTool.Commands.Settings;
+using Amazon.Lambda.TestTool.Extensions;
+using Amazon.Lambda.TestTool.Models;
+using Amazon.Lambda.TestTool.Services;
+
+namespace Amazon.Lambda.TestTool.Processes;
+
+/// <summary>
+/// A process that runs the API Gatewat emulator.
+/// </summary>
+public class ApiGatewayEmulatorProcess
+{
+    /// <summary>
+    /// The service provider that will contain all the registered services.
+    /// </summary>
+    public required IServiceProvider Services { get; init; }
+
+    /// <summary>
+    /// The API Gatewat emulator task that was started.
+    /// </summary>
+    public required Task RunningTask { get; init; }
+
+    /// <summary>
+    /// The endpoint of the API Gatewat emulator.
+    /// </summary>
+    public required string ServiceUrl { get; init; }
+
+    /// <summary>
+    /// Creates the Web API and runs it in the background.
+    /// </summary>
+    public static ApiGatewayEmulatorProcess Startup(RunCommandSettings settings, CancellationToken cancellationToken = default)
+    {
+        var builder = WebApplication.CreateBuilder();
+
+        builder.Services.AddApiGatewayEmulatorServices();
+        
+        var serviceUrl = $"http://{settings.Host}:{settings.ApiGatewayEmulatorPort}";
+        builder.WebHost.UseUrls(serviceUrl);
+        builder.WebHost.SuppressStatusMessages(true);
+
+        builder.Services.AddHealthChecks();
+
+        var app = builder.Build();
+
+        app.UseHttpsRedirection();
+
+        app.MapHealthChecks("/health");
+
+        app.Map("/{**catchAll}", (HttpContext context, IApiGatewayRouteConfigService routeConfigService) =>
+        {
+            var routeConfig = routeConfigService.GetRouteConfig(context.Request.Method, context.Request.Path);
```

Comment:
since routeconfig is already here, thats the reason i didnt try to access it again my translator function originally. we can discuss if we want to pass it in or if i just call the function that you made

Reply:
we talked offline on this. i will pass routeconfig into my function